lib/node_modules/@stdlib/repl/lib/main.js

```diff
@@ -321,6 +325,10 @@ function REPL( options ) {
 	// Set the syntax highlighting theme...
 	this.settings( 'theme', opts.settings.theme );
 
+	// Initialize specified bracketed-paste setting in TTY environments:
+	if ( this._isTTY ) {
+		this.settings( 'bracketedPaste', opts.settings.bracketedPaste );
+	}
```

Comment:
Should setting this setting happen always? Meaning, do we need the `_isTTY` guard here? Currently, we don't document that, when not TTY, we always ignore this setting. Not clear to me that this guard is necessary.

Reply:
I thought maybe we don't want to be writing any escape sequences to the output when things are not TTY. If the setting is `false`, we still are printing the escape sequences that disable bracketed paste, but if not TTY, I was trying to avoid writing anything at all to the ostream.

Reply:
You could make `autoDisableBracketedPasteOnExit` dependent on both `isTTY` AND whether bracketed paste was enabled. If a user explicitly opts in to bracketed paste, even when not in TTY, my sense is that, similar to syntax highlighting, we should honor the request and write escape sequences.

Reply:
The problem is the default behavior. Take the case where the user hasn't specified anything and is in a non-TTY context. Without the above check, we would still end up writing the escape sequences for "disabling" bracketed paste. That was my concern.

Reply:
The default value for `autoDisableBracketedPasteOnExit` should still remain whether is `isTTY`. It's only when exiting that we can make disabling dependent on whether bracketed paste is currently enabled. If already disabled, nothing to do.

Reply:
Re: default behavior. Ah, right. You could also make it a three-state setting. Namely, leave equal to `void 0`, rather than `false` when not TTY.

Reply:
...and only when explicitly set write escape sequences to the output stream.

Reply:
I think there's some confusion. I was referring to the initialization code that you have selected above. The `autoDisableBracketedPasteOnExit` is fine as it is. During REPL instantiation, depending on the setting of `bracketedPaste`, we need to `disable` or `enable` bracketed paste by printing the required escape sequences. In non-TTY contexts, the setting `bracketedPaste` gets the default of `false`, meaning we would still end up printing the escape sequences to disable bracketed paste even in non-TTY contexts. This was my main concern, that should we be printing anything in non-TTY contexts? I thought no, and hence the check.

Reply:
Okay, just now read your newer responses